validation/immersed_boundaries/flow_over_hills.jl

```diff
@@ -0,0 +1,160 @@
+using Printf
+using Statistics
+using Oceananigans
+using Oceananigans.ImmersedBoundaries: ImmersedBoundaryGrid, GridFittedBottom, mask_immersed_field!
+
+@inline bottom_drag_func(x, y, t, u, Cᴰ) = - Cᴰ * u^2
+
+function hilly_simulation(;
+                          Nx = 64,
+                          Nz = Nx,
+                          ϵ = 0.1,
+                          Re = 1e4,
+                          N² = 1e-2,
+                          bottom_drag = false,
+                          stop_time = 1,
+                          save_interval = 0.1,
+                          architecture = CPU(),
+                          name = "flow_over_hills")
+
+    underlying_grid = RectilinearGrid(architecture,
+                                      size = (Nx, Nz),
+                                      halo = (3, 3),
+                                      x = (0, 2π),
+                                      z = (0, 2π),
+                                      topology = (Periodic, Flat, Bounded))
+
+    if ϵ > 0
+        x, y, z = nodes((Center, Center, Center), underlying_grid, reshape=true)
+        hills = @. ϵ * (1 + sin(x)) / 2
+        grid = ImmersedBoundaryGrid(underlying_grid, GridFittedBottom(hills))
+    else # no hills
+        grid = underlying_grid
+    end
+
+    closure = isfinite(Re) ? ScalarDiffusivity(ν=1/Re, κ=1/Re) : nothing
+
+    if bottom_drag
+        Δz = 2π / Nz
```

Comment:
Maybe use `grid.Δzaac` here to be more general?

Reply:
Hmm, I feel this is a qualitative test so it doesn't matter?

Reply:
But ok.

Reply:
But wait. Using `grid.Δzᵃᵃᶜ` isn't generic because it doesn't work on a stretched grid. For things to be "correct" we need to do a lot more work (eg evaluate spacing following the hills, since spacing would change esp with `PartialBottomGrid`. Do you mind if we just keep the simplicity of this? We can do the hard work of generalizing it later if we need that.

Reply:
It's more in case you change the height on the domain down the line. But it's up you!

Reply:
Ok, lets plan to generalize this script when we need to!